v3/exchange.go

```diff
@@ -87,7 +88,33 @@ func (e *Exchange) heartbeat() {
 			select {
 			case <-ticker.C:
 				if _, err := e.Client().GetPublicTest(&operations.GetPublicTestParams{}); err != nil {
-					e.stop <- true
+
+					// We lost the connection so we attempt to re-connect
+					c, _, err := websocket.DefaultDialer.Dial(e.url, nil)
+					if err != nil {
+						log.Printf("Error in the default dialer %v", err)
```

Comment:
Add a method on exchange to allow users to redirect the output of this if they want.

e.g.

```
func (e *Exchange) SetLogOutput(w io.Writer) {
    log.SetOutput(w)
} 
```

This way users can redirect the log output to an existing logrus (or other logging package) instance.